crates/proof-of-sql/src/sql/proof_exprs/range_check.rs

```diff
@@ -23,44 +116,315 @@ fn decompose_scalar_to_words<'a, S: Scalar + 'a>(
+    for i in 0..31 {
+        let byte_column = &mut word_columns[i];
+
+        // Allocate words_plus_alpha
+        let words_plus_alpha: &mut [S] =
+            alloc.alloc_slice_fill_with(byte_column.len(), |j| S::from(&byte_column[j]) + alpha);
+
+        // Allocate words_plus_alpha_inv
+        let words_plus_alpha_inv: &mut [S] =
+            alloc.alloc_slice_fill_with(byte_column.len(), |j| S::from(&byte_column[j]) + alpha);
+        slice_ops::batch_inversion(&mut words_plus_alpha_inv[..]);
+
+        // Produce an MLE over words_plus_alpha
+        builder.produce_intermediate_mle(words_plus_alpha as &[_]);
+
+        // Copy words_plus_alpha_inv to the corresponding inverted_word_columns[i]
+        builder.produce_intermediate_mle(words_plus_alpha_inv as &[_]);
+
+        inverted_word_columns[i].copy_from_slice(words_plus_alpha_inv);
+        assert_eq!(words_plus_alpha_inv, inverted_word_columns[i]);
+
+        let input_ones = alloc.alloc_slice_fill_copy(table_length, true);
+
+        builder.produce_sumcheck_subpolynomial(
+            SumcheckSubpolynomialType::Identity,
+            vec![
+                (
+                    S::one(),
+                    vec![
+                        Box::new(words_plus_alpha as &[_]),
+                        Box::new(words_plus_alpha_inv as &[_]),
+                    ],
+                ),
+                (-S::one(), vec![Box::new(input_ones as &[_])]),
+            ],
+        );
+    }
+}
+
+/// Produce the range of possible values that a word can take on,
+/// based on the word's bit size, along with an intermediate MLE:
+///
+/// ```text
+/// | Column 0           |
+/// |--------------------|
+/// |  0                 |
+/// |  1                 |
+/// |  ...               |
+/// |  2ⁿ - 1            |
+/// ----------------------
+///       |       
+///       v  
+///    Int. MLE
+/// ```
+/// Here, `n` represents the bit size of the word (e.g., for an 8-bit word, `2⁸ - 1 = 255`).
+///
+/// Then, add the verifier challenge α, invert, and produce an
+/// intermediate MLE:
+///
+/// ```text
+/// | Column 0
+/// |--------------------|
+/// | (0 + α)⁻¹          |
+/// | (1 + α)⁻¹          |
+/// | ...                |
+/// | (2ⁿ - 1 + α)⁻¹     |
+/// ----------------------
+///       |      
+///       v        
+///    Int. MLE  
+/// ```
+/// Finally, argue that (`word_values` + α)⁻¹ * (`word_values` + α) - 1 = 0
+///
+use alloc::vec;
+#[allow(clippy::missing_panics_doc)]
+#[allow(clippy::cast_possible_truncation)]
+fn prove_word_values<'a, S: Scalar + 'a>(
+    alloc: &'a Bump,
+    scalars: &[S],
+    alpha: S,
+    table_length: usize,
+    builder: &mut FinalRoundBuilder<'a, S>,
+) {
+    // Allocate from 0 to 255 and pertrub with verifier challenge
+    let word_values_plus_alpha: &mut [S] =
+        alloc.alloc_slice_fill_with(max(256, scalars.len()), |i| S::from(&(i as u8)) + alpha);
+    builder.produce_intermediate_mle(word_values_plus_alpha as &[_]);
```

Comment:
Since the prover is going to compute the commitment of this, it would be far more efficient to simply commit to the words, rather than the word+alpha. This is because the words are small (8 bits), but words+alpha are large (~250 bits).

Reply:
This comment is unclear. I understand you are saying it is better to commit to the words vs words plus alpha, what is your suggestion here? Are you saying to not commit to word values at all? Commit to word values only and not values + alpha? (this causes a verification failure), Remove this line entirely?

To be clear, this is not a column of words as instantiated elsewhere in this code, this is a column of word values that the words take on, which in the case of byte sized words, are the values from 0..255. Originally we had this committed to as an anchored MLE because the verifier trivially has access to the values 0..255 if I remember correctly. Not sure if it makes sense to go back to that.